cmd/secrets-config/README.md

```diff
@@ -22,95 +22,61 @@ Proxy configuration commands (listed below) require access to the secret store m
-      This value must match the configured authentication plugin
-      (`KongAuth.Name` in security-proxy-setup's `configuration.toml`).
+    Create an API gateway user by creating a user identity the EdgeX secret store.
+    Requires additional arguments:
 
     * **--user** _username_ (required)
 
       Username of the user to add.
 
-    * **--group** _group_ (optional)
-
-      Group to which the user belongs, defaults to &quot;admin&quot;.
-      This should be the group associated with the route ACL
-      (`KongAuth.WhiteList` in security-proxy-setup's `configuration.toml`).
-      (Note that secrets-config shares the same configuration as security-proxy-setup
-      as they both configure the EdgeX API gateway.)
-
-
-    The following options are used when token-type == "jwt":
-
-    * **--algorithm** RS256 | ES256 (required for JWT method)
-
-      Algorithm used for signing the JWT.
-      (See [RFC 7518](https://tools.ietf.org/html/rfc7518#section-3.1) for a list of signing algorithms.)
+    * **--useRootToken** (optional)
 
-    * **--public\_key** _/path/to/public\_key_ (required for JWT tokens)
+      Normally, `secrets-config` uses a service token in the secret store token file.
+      As this token expires from inactivity an hour after it is created,
+      it is possible to point `secrets-config` at a `resp-init.json`
+      and a root token will be created afresh from the key shares in that file.
+      The `--useRootToken` flag is used to tell `secrets-config`
+      to use this authentication method to talk to the EdgeX secret store.
 
-      Public key (in PEM format) used to validate the JWT.
-      (Not an x.509 certificate.)
-      This key is assumed to have been pre-created using some external mechanism such as a TPM, HSM, openssl, or other method.
+    Upon completion, `adduser` returns a JSON object with a random `password` field set.
+    This password is generated from the kernel random source and overwrites any previous password set on the account.
 
-    * **--id** _key_ (optional)
+    A sample shell script to turn this into an token that can be used for API gateway authentication is as follows:
 
-      Optional user-specified &quot;key&quot; used for linkage with an incoming JWT via Kong&#39;s config.key\_claim\_name setting (defaults to &quot;iss&quot; field).
-      See
-      [Kong documentation for JWT plugin](https://docs.konghq.com/hub/kong-inc/jwt/#craft-a-jwt-with-publicprivate-keys-rs256-or-es256)
-      for an example of how this parameter is used.
+    ```shell
+    password=password-from-above
 
-    Upon completion, for token-type == "jwt", the command outputs the autogenerated _key_ for the **id** command above.
-    This value must be used during later construction of the JWT.
+    vault_token=$(curl -ks "http://localhost:8200/v1/auth/userpass/login/${username}" -d "{\"password\":\"${password}\"}" | jq -r '.auth.client_token')
```

Comment:
what happens if this fails, the developer can re-run or restart the service?

Reply:
Vault is always running.  Additional runs of secrets-config adduser will set a new password on the account at which point need to go through the login/generate jwt process again.

Reply:
could you add what you said into docs to help developers in this case? thanks!

Reply:
Added that as well as other documentation.